modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptMethodsGenerator.cs

```diff
@@ -135,7 +136,9 @@ INamedTypeSymbol symbol
 
             source.Append("#pragma warning disable CS0109 // Disable warning about redundant 'new' keyword\n");
 
-            source.Append($"    public new class MethodName : {symbol.BaseType.FullQualifiedName()}.MethodName {{\n");
+            Debug.Assert(symbol.BaseType != null, "Symbol always has a base type because every script class must inherit from a Godot type; otherwise, the generator wouldn't be visiting this class.");
```

Comment:
I really don't like the fact flow analysis can't understand the value cannot be null after this check in .NET Standard 2.0. I think throwing an exception would be better.

Reply:
Actually, wouldn't this check be excluded when publishing the package?

```cs
[System.Diagnostics.Conditional("DEBUG")]
public static void Assert (bool condition);
```

Calls to this function are only compiled in Debug configurations. It's the same as:

```cs
#if DEBUG
Debug.Assert(foo);
#endif
```

Reply:
Yes, the call is excluded, that's why I'm using it in these cases because they will never be null so I wanted to avoid the unnecessary check but it might still be useful during development to notice if we're doing something wrong.